frontend/app/routes/_app+/new-application+/a2.tsx

```diff
@@ -0,0 +1,83 @@
+import { json, redirect } from '@remix-run/node';
+import type { ActionFunctionArgs, LoaderFunctionArgs } from '@remix-run/node';
+import { Form, useLoaderData } from '@remix-run/react';
+
+import { useTranslation } from 'react-i18next';
+
+import { Button, ButtonLink } from '~/components/buttons';
+import { InputRadios } from '~/components/input-radios';
+import { getRaoidcService } from '~/services/raoidc-service.server';
+import { getSessionService } from '~/services/session-service.server';
+import { getTypedI18nNamespaces } from '~/utils/locale-utils';
+
+const i18nNamespaces = getTypedI18nNamespaces('new-application');
+
+export const handle = {
+  breadcrumbs: [{ labelI18nKey: 'new-application:a2.breadcrumbs.new-application' }, { labelI18nKey: 'new-application:a2.breadcrumbs.a2' }],
+  i18nNamespaces,
+  pageIdentifier: 'CDCP-0016',
+  pageTitleI18nKey: 'new-application:a2.page-title',
+};
+
+export async function loader({ request }: LoaderFunctionArgs) {
+  const raoidcService = await getRaoidcService();
+  await raoidcService.handleSessionValidation(request);
+
+  const sessionService = await getSessionService();
+  const session = await sessionService.getSession(request);
+  const selectedType = session.get('selectedType');
+
+  return json({ selectedType });
+}
+
+export async function action({ request }: ActionFunctionArgs) {
+  const raoidcService = await getRaoidcService();
+  await raoidcService.handleSessionValidation(request);
+
+  const sessionService = await getSessionService();
+  const session = await sessionService.getSession(request);
+
+  const formDataRadio = Object.fromEntries(await request.formData());
+  const selectedType = formDataRadio.selectedType === 'self';
+  session.set('selectedType', selectedType);
+
+  return redirect('/new-application/a3', {
+    headers: {
+      'Set-Cookie': await sessionService.commitSession(session),
+    },
+  });
+}
+
+export default function A2() {
+  useLoaderData<typeof loader>();
+  const { t } = useTranslation(i18nNamespaces);
+  const updateValidationMessage = () => {
+    (document.getElementById('input-radio-selected-type-option-0') as HTMLInputElement).setCustomValidity(t('new-application:a2.required'));
+  };
```

Comment:
This is an interesting way to handle form validation, but it doesn't really follow the canada.ca way of doing it. I'm interested in what @sebastien-comeau and @nicholas-ly think about this.

Also.. this approach doesn't survive a language toggle, which is pretty important (in my opinion).

Reply:
Yup, TIL but it has some complexity since we are managing errors on the route's action. Even the GCWeb example doesn't override the default browser error messages on the client.
https://wet-boew.github.io/gcweb-compiled-demos/wetboew-demos/formvalid/formvalid-fr.html

Reply:
Admittedly, leaving the browser error message untranslated seems inconsistent to me, but if that's the standard... ¯\\_(ツ)_/¯ 

Now removed.

Reply:
When handling the form validation in the Remix route, the browser error message isn't triggered, so there's no need to worry about its translation. Look at how error handling is done in <https://github.com/DTS-STN/canadian-dental-care-plan/blob/main/frontend/app/routes/_app%2B/personal-information%2B/phone-number%2B/edit.tsx>

![image](https://github.com/DTS-STN/canadian-dental-care-plan/assets/48123208/dd9df9b2-a546-40b5-8ecd-230fd33a28bf)

Reply:
![Screenshot from 2024-02-28 07-03-24](https://github.com/DTS-STN/canadian-dental-care-plan/assets/154452024/c505662d-edc6-42f3-a095-7c5465fd8a9b)

This looks like the browser error message is being triggered, I don't see that error message anywhere in the project.

Reply:
You're right. I think we should fix this in all of our forms. It's not the best user experience.